tests/functional/adapter/test_incremental_iceberg.py

```diff
@@ -0,0 +1,232 @@
+"""
+Run optional dbt functional tests for Iceberg incremental merge, including delete and incremental predicates.
+
+"""
+import re
+
+import pytest
+
+from dbt.contracts.results import RunStatus
+from dbt.tests.adapter.incremental.test_incremental_predicates import (
+    BaseIncrementalPredicates,
+)
+from dbt.tests.adapter.incremental.test_incremental_unique_id import (
+    BaseIncrementalUniqueKey,
+    models__duplicated_unary_unique_key_list_sql,
+    models__empty_str_unique_key_sql,
+    models__empty_unique_key_list_sql,
+    models__expected__one_str__overwrite_sql,
+    models__expected__unique_key_list__inplace_overwrite_sql,
+    models__no_unique_key_sql,
+    models__nontyped_trinary_unique_key_list_sql,
+    models__not_found_unique_key_list_sql,
+    models__not_found_unique_key_sql,
+    models__str_unique_key_sql,
+    models__trinary_unique_key_list_sql,
+    models__unary_unique_key_list_sql,
+    seeds__add_new_rows_sql,
+    seeds__duplicate_insert_sql,
+    seeds__seed_csv,
+)
+from dbt.tests.util import run_dbt
+
+seeds__expected_incremental_predicates_csv = """id,msg,color
+3,anyway,purple
+1,hey,blue
+2,goodbye,red
+2,yo,green
+"""
+
+seeds__expected_delete_condition_csv = """id,msg,color
+1,hey,blue
+3,anyway,purple
+"""
+
+seeds__expected_predicates_and_delete_condition_csv = """id,msg,color
+1,hey,blue
+1,hello,blue
+3,anyway,purple
+"""
+
+
+class TestIcebergMergeNonUniqueLocation:
```

Comment:
most likely my last comment got lost, could you move this to test_basic_iceberg? 🙏 as this is more generic than just incremental.

Reply:
@chrischin478 have a look this then the PR is ready to be merged.

Reply:
That test is now located in the `test_basic_iceberg.py` file here:

https://github.com/dbt-athena/dbt-athena/blob/0c95f9c6fd84b46c19b806d01620f8c9f08e3f7c/tests/functional/adapter/test_basic_iceberg.py#L79-L81